src/systems/physics/Physics.cc

```diff
@@ -1549,43 +1641,50 @@ void PhysicsPrivate::UpdateSim(EntityComponentManager &_ecm)
           auto canonicalLink =
               _ecm.Component<components::CanonicalLink>(_entity);
 
-          // get the pose component of the parent model
-          const components::Pose *parentPose =
-              _ecm.Component<components::Pose>(_parent->Data());
-
           auto frameData = linkIt->second->FrameDataRelativeToWorld();
           const auto &worldPose = frameData.pose;
 
-          // if the parentPose is a nullptr, something is wrong with ECS
-          // creation
-          if (!parentPose)
-          {
-            ignerr << "The pose component of " << _parent->Data()
-                   << " could not be found. This should never happen!\n";
-            return true;
-          }
           if (canonicalLink)
           {
-            // This is the canonical link, update the model
-            // The Pose component, _pose, of this link is the initial
-            // transform of the link w.r.t its model. This component never
-            // changes because it's "fixed" to the model. Instead, we change
-            // the model's pose here. The physics engine gives us the pose of
-            // this link relative to world so to set the model's pose, we have
-            // to post-multiply it by the inverse of the initial transform of
-            // the link w.r.t to its model.
-            auto mutableParentPose =
-              _ecm.Component<components::Pose>(_parent->Data());
-            *(mutableParentPose) = components::Pose(_pose->Data().Inverse() +
-                                           math::eigen3::convert(worldPose));
+            // This is the canonical link, update the top level model.
+            // The pose this link w.r.t its top level model never changes
```

Comment:
Thought might as well fix the typo here (pose 'of' this link)

Reply:
fixed. 7740d5cc86ab37ed863e13e377bb551367c2367b